src/GoodsList.tsx

```diff
@@ -2,15 +2,26 @@ import React from 'react';
 import { Good } from './types/Good';
 
 type Props = {
-  goods: Good[]
+  goods: Good[],
+  hasError: boolean,
 };
 
-export const GoodsList: React.FC<Props> = ({ goods }) => (
-  <ul>
-    {goods.map(good => (
-      <li key={good.id} data-cy="good">
-        {good.name}
-      </li>
-    ))}
-  </ul>
+export const GoodsList: React.FC<Props> = ({ goods, hasError }) => (
+  <>
+    { hasError
+      ? (<p>Error, server is unavailable</p>)
```

Comment:
I think it's better to move to display an error to the App component. GoodsList must display only a list.